src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo).append("\n");
+        LocalDate from = null;
+        LocalDate to = null;
+
+        try {
+            from = LocalDate.parse(dateFrom, formatter);
+            to = LocalDate.parse(dateTo, formatter);
+        } catch (DateTimeParseException e) {
+            System.out.println("Error parsing from & to Dates after initialize"
+                    + dateFrom + " "
+                    + dateTo + ". Error: "
+                    + e);
```

Comment:
The error message here is too generic and could be improved by providing more context. Consider specifying which date (from or to) failed to parse and include the problematic date string in the message.